lib/heathen/processor_methods/libreoffice.rb

```diff
@@ -48,19 +44,36 @@ def libreoffice( format: )
       end
       raise InvalidMimeTypeInStep.new('(various document formats)', job.mime_type) unless to_suffix
 
-      conversion_method = conversion_methods[to_suffix] || to_suffix
-      target_file       = "#{job.content_file}.#{to_suffix}"
-      executioner.execute(
-        'libreoffice',
-        '--convert-to', conversion_method,
-        '--outdir', sandbox_dir,
-        job.content_file,
-        '--headless',
-      )
+      target_file = "#{job.content_file}.#{to_suffix}"
+
+      if to_suffix == 'txt'
+        pdf_file       = "#{job.content_file}.pdf"
+        executioner.execute(
+          'libreoffice',
+          '--convert-to', 'pdf',
+          '--outdir', sandbox_dir,
+          job.content_file,
+          '--headless',
+        )
+
+        executioner.execute(
+          'pdftotext',
+          pdf_file,
+          target_file
```

Comment:
as well here `target_file` is misleading, it is used only here, so it's OK to do the concatenation of `job.content_file` with `to_suffix` here.